ports/rp2/machine_spi.c

```diff
@@ -261,61 +261,14 @@ static void machine_spi_init(mp_obj_base_t *self_in, size_t n_args, const mp_obj
 
 static void machine_spi_transfer(mp_obj_base_t *self_in, size_t len, const uint8_t *src, uint8_t *dest) {
     machine_spi_obj_t *self = (machine_spi_obj_t *)self_in;
-    // Use DMA for large transfers if channels are available
-    const size_t dma_min_size_threshold = 32;
-    int chan_tx = -1;
-    int chan_rx = -1;
-    if (len >= dma_min_size_threshold) {
```

Comment:
Considering the bug can only (so far?) be reproduced on boards with PSRAM, maybe the best approach is to change the logic here so that it never uses DMA if `src` or `dest` point to PSRAM?

Reply:
OK, I see.

I'll have to think about this more, but in general I think it's worth keeping the SPI DMA code.  So how about making the logic be "don't use DMA if on an RP2350 with PSRAM"?

Reply:
Fair enough, I can probably wrap the DMA code in an #ifdef with the relevant PSRAM macro (not at my computer right now).

Just curious, what's the motivation for using DMA channels for SPI transfers?